ui/app/src/components/confirmationDialog/CreatePoolConfirmationDialog.vue

```diff
@@ -0,0 +1,221 @@
+<script lang="ts">
+import { PropType, defineComponent, ref } from "vue";
+import { computed } from "@vue/reactivity";
+import ConfirmationModal, {
+  ConfirmState,
+} from "@/components/shared/ConfirmationModal.vue";
+import DetailsPanelPool from "@/components/shared/DetailsPanelPool.vue";
+import AssetItemLarge from "@/components/shared/AssetItemLarge.vue";
+import { useAssetItem } from "@/components/shared/utils";
+
+export default defineComponent({
+  components: {
+    ConfirmationModal,
+    DetailsPanelPool,
+    AssetItemLarge,
+  },
+
+  props: {
+    isOpen: { type: Boolean, default: false }, 
+    requestClose: Function,
+    state: { type: String as PropType<ConfirmState>, default: "confirming" },
+    transactionHash: String,
+    transactionStateMsg: String,
+    fromToken: String,
+    fromAmount: String,
+    poolUnits: String,
+    toAmount: String,
+    toToken: String,
+    aPerB: String,
+    bPerA: String,
+    shareOfPool: String,
+  },
+
+  setup(props, { emit }) {
+    const fromSymbol = computed(() => props.fromToken);
+    const fromAsset = useAssetItem(fromSymbol);
+    const fromLabel = fromAsset.label;
+    const fromImage = computed(() => fromAsset.token.value ? fromAsset.token.value.imageUrl : '');
+    const fromAmount = computed(() => props.fromAmount);
+
+    const toSymbol = computed(() => props.toToken);
+    const toAsset = useAssetItem(toSymbol);
+    const toLabel = toAsset.label;
+    const toImage = computed(() => toAsset.token.value ? toAsset.token.value.imageUrl : '');
+    const toAmount = computed(() => props.toAmount);
```

Comment:
@ryardley I might be wrong, but it looks like excessive reactivity here. we have reactive props outside and then also use models and events. I think we should debug how many times we reload components with one action to make sure that all is good

Reply:
yes props are already reactive.